cli/config/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+//! Manages tree-sitter's configuration file.
+
+use anyhow::{anyhow, Result};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::path::PathBuf;
+use std::{env, fs};
+
+/// Holds the contents of tree-sitter's configuration file.
+///
+/// The file typically lives at `~/.config/tree-sitter/config.json`, but see the [`load`][] method
+/// for the full details on where it might be located.
+///
+/// This type holds the generic JSON content of the configuration file.  Individual tree-sitter
+/// components will use the [`get`][] method to parse that JSON to extract configuration fields
+/// that are specific to that component.
+pub struct Config {
+    location: PathBuf,
+    config: Value,
+}
+
+impl Config {
+    fn find_config_file() -> Result<PathBuf> {
+        if let Ok(path) = env::var("TREE_SITTER_DIR") {
+            let mut path = PathBuf::from(path);
+            path.push("config.json");
+            return Ok(path);
+        }
+
+        let xdg_path = Self::xdg_config_file()?;
+        if xdg_path.is_file() {
+            return Ok(xdg_path);
+        }
+
+        let legacy_path = dirs::home_dir()
+            .ok_or(anyhow!("Cannot determine home directory"))?
+            .join(".tree-sitter/config.json");
+        if legacy_path.is_file() {
+            return Ok(legacy_path);
+        }
+
+        Err(anyhow!(concat!(
+            "Cannot find a tree-sitter configuration file.\n",
+            "Please run `tree-sitter init-config` to create one!"
+        )))
```

Comment:
Note that this makes it a hard error to run `tree-sitter` without having a config file in place, which is different than before.  I'm honestly not sure that I like this part of the change.

Reply:
Ha! And in fact that makes the test cases fail!  Okay that definitely tells me we don't want this to be a hard error.  I'll push up a fix for that.